app/client/src/components/RoleGuard.tsx

```diff
@@ -0,0 +1,14 @@
+import router from 'next/router';
+
+export interface RoleGuardProps {
+    organizer?: boolean;
+    moderator?: boolean;
```

Comment:
Could you add some simple comments here? For example, `true if the moderator is an organizer`.  React props kinda breaks best practice re: naming booleans. Normally, you'd want `isOrganizer`, but with React props booleans are typicallly just a noun. 

In summary, what you have is correct, please just add some basic comments.